frontend/src/scenes/error-tracking/queries.ts

```diff
@@ -1,47 +1,109 @@
 }): DataTableNode => {
+    const { value, displayAs, offsetHours } = parseSelection(sparklineSelection)
+
+    const { labels, data } = generateSparklineProps({ value, displayAs, offsetHours })
+
     return {
         kind: NodeKind.DataTableNode,
         source: {
             kind: NodeKind.EventsQuery,
             select: [
-                'any(properties) -- Error',
+                'any(properties) as "context.columns.error"',
                 'properties.$exception_type',
-                `sparkline(reverse(arrayMap(x -> countEqual(groupArray(dateDiff('hour', now() - INTERVAL 1 day, timestamp)), x), range(24)))) -- Volume`,
-                'count() as unique_occurrences -- Occurrences',
-                'count(distinct $session_id) as unique_sessions -- Sessions',
-                'count(distinct distinct_id) as unique_users -- Users',
+                `<Sparkline data={${data}} labels={[${labels.join(',')}]} /> as "context.columns.volume"`,
+                'count() as occurrences',
+                'count(distinct $session_id) as sessions',
+                'count(distinct distinct_id) as users',
                 'max(timestamp) as last_seen',
                 'min(timestamp) as first_seen',
             ],
             orderBy: [order],
             ...defaultProperties({ dateRange, filterTestAccounts, filterGroup }),
         },
-        hiddenColumns: [
-            'properties.$exception_type',
-            'first_value(properties)',
-            'max(timestamp) as last_seen',
-            'min(timestamp) as first_seen',
-        ],
+        hiddenColumns: ['$exception_type', 'last_seen', 'first_seen'],
         showActions: false,
         showTimings: false,
+        columns: [
+            'context.columns.error',
+            '$exception_type',
+            'context.columns.volume',
+            'occurrences',
+            'sessions',
+            'users',
+            'last_seen',
+            'first_seen',
+        ],
     }
 }
 
+const parseSelection = (selection: SparklineOption): ErrorTrackingSparklineConfig => {
+    if (selection.value in SPARKLINE_CONFIGURATIONS) {
+        return { ...selection, ...SPARKLINE_CONFIGURATIONS[selection.value] }
+    }
+
+    const [value, unit] = selection.value.replace('-', '').split('')
+
+    return {
+        ...selection,
+        value: Number(value),
+        displayAs: unit === 'm' ? 'minute' : unit === 'h' ? 'hour' : 'day',
+    }
+}
+
+export const generateSparklineProps = ({
+    value,
+    displayAs,
+    offsetHours,
+}: ErrorTrackingSparklineConfig): { labels: string[]; data: string } => {
+    const offset = offsetHours ?? 0
+    const now = dayjs().subtract(offset, 'hour').startOf(displayAs)
+    const dates = range(value).map((idx) => now.subtract(value - (idx + 1), displayAs))
+    const labels = dates.map((d) => `'${d.format('D MMM, YYYY HH:mm')} (UTC)'`)
+
+    const startTime = `subtractHours(now(), ${offset})`
+    const toStartOfIntervalFn =
```

Comment:
Sadly we don't support `toStartOfInterval` in HogQL which would be really useful to group by periods other than what we're searching over e.g. For "Last 7 days"
- Grouped as days has too few bars (7) to look good
- Grouped by hour (7*24 = 168) doesn't look great by it's what we went with
- Grouped in 8 hour chunks gives 21 bars which looks the best but requires `toStartOfInterval(timestamp, INTERVAL 8 HOUR)`

Reply:
7 days looks good in the gif

i guess maybe we ship this and you can follow-up adding `toStartOfInterval`?